src/python/bcc/table.py

```diff
@@ -457,25 +528,22 @@ def items_delete_batch(self, keys=None):
             for _ in self.items_lookup_and_delete_batch():
                 return
 
-    def items_update_batch(self, keys, values):
+    def items_update_batch(self, ct_keys, ct_values):
         """Update all the key-value pairs in the map provided.
-        The lists must be the same length, between 1 and the maximum number of entries.
+        The arrays must be the same length, between 1 and the maximum number
+        of entries.
+
+        Args:
+            ct_keys (ct.Array): keys array to update
+            ct_values (ct.Array): values array to update
+        Raises:
+            Exception: If bpf syscall has failed or is invalid
         """
-        # two ct.Array are expected
-        if not isinstance(keys, ct.Array) or not isinstance(values, ct.Array):
-            raise TypeError
-
-        batch_size = len(keys)
-
-        # check that batch between limits and coherent with the provided values
-        if batch_size < 1 or batch_size > self.max_entries or batch_size != len(values):
-            raise KeyError
-
-        count = ct.c_uint32(batch_size)
+        ct_cnt = self._sanity_check_keys_values(keys=ct_keys, values=ct_values)
         res = lib.bpf_update_batch(self.map_fd,
-                                   ct.byref(keys),
-                                   ct.byref(values),
-                                   ct.byref(count)
+                                   ct.byref(ct_keys),
+                                   ct.byref(ct_values),
+                                   ct.byref(ct_cnt)
                                    )
 
         errcode = ct.get_errno()
```

Comment:
similar to delete_batch, just check whether errcode is 0 or not is enough below.